contract/src/agoric-basics.contract.js

```diff
@@ -101,39 +154,45 @@ export const start = async zcf => {
   /** @type {OfferHandler} */
   const tradeHandler = buyerSeat => {
     // give and want are guaranteed by Zoe to match proposalShape
-    const { want } = buyerSeat.getProposal();
+    const { give, want } = buyerSeat.getProposal();
+
+    // check that we have enough inventory
+    AmountMath.isGTE(
+      inventorySeat.getCurrentAllocation().Tickets,
+      want.Tickets,
+    ) || Fail`Not enough inventory, ${q(want.Tickets)} wanted`;
 
-    sum(bagCounts(want.Items.value)) <= maxItems ||
-      Fail`max ${q(maxItems)} items allowed: ${q(want.Items)}`;
+    // check that user is paying enough for all the tickets
+    const totalPrice = bagPrice(want.Tickets.value, inventory);
+    AmountMath.isGTE(give.Price, totalPrice) ||
+      Fail`Total price is ${q(totalPrice)}, but ${q(give.Price)} was given`;
 
-    const newItems = itemMint.mintGains(want);
     atomicRearrange(
       zcf,
       harden([
         // price from buyer to proceeds
-        [buyerSeat, proceeds, { Price: tradePrice }],
-        // new items to buyer
-        [newItems, buyerSeat, want],
+        [buyerSeat, proceeds, { Price: totalPrice }],
+        // tickets from inventory to buyer
+        [inventorySeat, buyerSeat, want],
       ]),
     );
 
     buyerSeat.exit(true);
```

Comment:
_question, for my learning_

What to do with `proceeds` and `inventorySeat`?

Seems like we could get the proceeds when contract is closed by the creator via creatorFacet. Should we keep `inventorySeat` open even in the case when all tickets are sold?

Reply:
yes, a `creatorFacet` method is a typical way to collect the proceeds. (Seems fine to postpone to a later PR, if we bother to do it at all)
See, for example, [collectFees.js](https://github.com/Agoric/agoric-sdk/blob/b1176991359ebd7bfaf858956b8648b1d88e80cf/packages/inter-protocol/src/collectFees.js) and the contracts that import it.

as to `inventorySeat`... when they're all sold, we could shut the contract down (there's a shutdown method on zcf)

Or we could have a creatorFacet method to resupply it for the next event or something. But that seems like over-kill.